data/resources/ui/content-message-document.ui

```diff
@@ -0,0 +1,77 @@
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
I noticed that the message is visually inconsistent if there is a caption. You need one more vertical box at the root. 

Before
![before](https://user-images.githubusercontent.com/3630213/189496772-a82541d6-48fd-4268-b30c-ccaaea4ec608.png)

After
![after](https://user-images.githubusercontent.com/3630213/189496770-427ff4f0-37d5-4443-9db8-93bb0398968c.png)

Here is the whole file. Keep in mind that you also have to update `style.css`.
```xml
<?xml version="1.0" encoding="UTF-8"?>
<interface>
  <template class="ContentMessageDocument" parent="ContentMessageBase">
    <property name="layout-manager">
      <object class="GtkBoxLayout">
        <property name="orientation">vertical</property>
      </object>
    </property>
    <style>
      <class name="message-bubble"/>
    </style>
    <child>
      <object class="GtkLabel" id="sender_label">
        <property name="ellipsize">end</property>
        <property name="single-line-mode">True</property>
        <property name="xalign">0</property>
        <style>
          <class name="sender-text"/>
        </style>
      </object>
    </child>

    <child>
      <object class="GtkBox">
        <property name="orientation">vertical</property>

        <child>
          <object class="GtkBox">
            <property name="spacing">6</property>
            <child>
              <object class="GtkButton" id="file_status">
                <style>
                  <class name="circular"/>
                </style>
                <property name="valign">center</property>
              </object>
            </child>
            <child>
              <object class="GtkBox">
                <property name="valign">center</property>
                <property name="hexpand">true</property>
                <property name="orientation">vertical</property>
                <child>
                  <object class="GtkLabel" id="file_name">
                    <property name="xalign">0</property>
                    <property name="ellipsize">middle</property>
                  </object>
                </child>
                <child>
                  <object class="GtkLabel" id="file_size">
                    <style>
                      <class name="numeric"/>
                      <class name="dim-label"/>
                    </style>
                    <property name="xalign">0</property>
                  </object>
                </child>

              </object>
            </child>
          </object>
        </child>

        <child>
          <object class="MessageIndicators">
            <property name="message" bind-source="indicators" bind-property="message"/>
            <property name="visible" bind-source="content_label" bind-property="visible" bind-flags="sync-create|invert-boolean"/>
            <property name="halign">end</property>
            <property name="valign">end</property>
          </object>
        </child>

      </object>
    </child>

    <child>
      <object class="MessageLabel" id="content_label">
        <property name="indicators">
          <object class="MessageIndicators" id="indicators">
            <property name="halign">end</property>
            <property name="valign">end</property>
          </object>
        </property>
      </object>
    </child>
  </template>
</interface>
```

Reply:
I think setting button valign to start is enough because i can't see other differences
![image1](https://user-images.githubusercontent.com/63008755/189497325-191c7837-3c55-4927-9208-c02eb7a9489b.png)

Reply:
Yeah, that's even better :-)

Reply:
An even better solution would be to make the indicators an overlay of the entire widget, and then removing it when there’s a caption (similar to what MessageMedia does).

But actually I’ve noticed that this patterns is shared for other message types as well (like voice messages, videos, animations) so I think that it could make sense to create a sort of bin widget to share this handling. I’ll work on that in the future, for now this implementation is fine.

Reply:
I think it's need to add prefix and suffix widgets to regular message, and then put content as prefix (documents, images, replies) or suffix (link preview)

It would be useful for grouping documents 

Looks like [unigram](https://github.com/UnigramDev/Unigram/tree/develop/Unigram/Unigram/Controls/Messages) does something like that

Reply:
Yeah I was thinking to implement something like that. I was also thinking to manage the sender in that widget so that we can also remove that code duplication.